chanzuckerberg/single-cell-data-portal
# fix: meta endpoint correctly looks up replaced datasets

https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/single-cell-data-portal/4192

## Changes

- Retrieve the correct dataset using the source of truth from the collection mappings.

## Testing steps

- Added a unit test specifically for this case.
- Since this cannot be easily tested in rdev, my recommendation is to do a prod->dev mirroring and use [this collection](https://cellxgene.cziscience.com/collections/f5af7a2f-ab4c-4728-829e-48efb9562105) - which prompted the issue - to verify the correct behavior.
